spec/services/sign_in/openid_connect_certificates_presenter_spec.rb

```diff
@@ -6,12 +6,31 @@
   describe '#perform' do
     subject { SignIn::OpenidConnectCertificatesPresenter.new.perform }
 
-    let(:public_key_jwks) { { keys: [public_key_jwk] } }
-    let(:public_key_jwk) {  JWT::JWK.new(public_key, { alg: 'RS256', use: 'sig' }).export }
+    let(:public_key_jwk) { JWT::JWK.new(public_key, { alg: 'RS256', use: 'sig' }).export }
     let(:public_key) { OpenSSL::PKey::RSA.new(File.read(Settings.sign_in.jwt_encode_key)).public_key }
 
-    it 'returns JWK representation of sign in service access token encode key' do
-      expect(subject).to eq(public_key_jwks)
+    shared_examples 'certificates return' do
+      before { Settings.sign_in.jwt_old_encode_key = old_encode_key_setting }
+
+      it 'returns a JWK representation of Sign in Service public keys' do
+        expect(subject).to eq(expected_public_keys_jwks)
+      end
+    end
+
+    context 'without an old Sign in Service access token encode setting' do
+      let(:expected_public_keys_jwks) { { keys: [public_key_jwk] } }
+      let(:old_encode_key_setting) { nil }
+
+      it_behaves_like 'certificates return'
+    end
+
+    context 'with an old Sign in Service access token encode setting' do
+      let(:expected_public_keys_jwks) { { keys: [public_key_jwk, old_public_key_jwk] } }
+      let(:old_public_key_jwk) { JWT::JWK.new(old_public_key, { alg: 'RS256', use: 'sig' }).export }
+      let(:old_public_key) { OpenSSL::PKey::RSA.new(File.read(Settings.sign_in.jwt_old_encode_key)) }
+      let(:old_encode_key_setting) { Settings.sign_in.jwt_old_encode_key || Settings.sign_in.jwt_encode_key }
```

Comment:
which `Settings` are you meaning to set this to here? You should remove conditional logic in `let` values and represent that in `contexts` if necessary

Reply:
The intent was to make the test function with the (zero currently) instances that have a `jwt_old_encode_key` while not breaking for the instances that don't have that setting. I just changed it to duplicate the `jwt_encode_key` as a faux-old key in all settings to keep it consistent.